MultiModalCLF/MoodBoardMultiModalClassification.ipynb

```diff
@@ -0,0 +1,3617 @@
+              "version_major": 2,
+              "version_minor": 0,
+              "model_id": "3e3afeed03384254be9d5e8d32c7cd73"
+            }
+          },
+          "metadata": {}
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "data = pd.read_csv('trimmed_valid_images.csv')\n",
+        "train_set, test_set = train_test_split(data, test_size=0.2, random_state=42)  # 20% for testing\n",
+        "train_set.to_csv('train_data.csv', index=False)\n",
+        "test_set.to_csv('test_data.csv', index=False)"
+      ],
+      "metadata": {
+        "id": "R0jr2NkEQ_IG"
+      },
+      "execution_count": 10,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Dataset objects to feed into trainer\n",
+        "train_dataset = MoodBoardDataset(csv_file='train_data.csv')\n",
+        "test_dataset = MoodBoardDataset(csv_file='test_data.csv')"
+      ],
+      "metadata": {
+        "id": "iQIPhbjlWY67"
+      },
+      "execution_count": 11,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "minidata = pd.read_csv('test_data.csv')\n",
+        "minitrain, minitests = train_test_split(minidata, test_size=0.3, random_state=42)\n",
+        "minitrain.to_csv('minitrain.csv', index=False)\n",
+        "minitests.to_csv('minitests.csv', index=False)"
+      ],
+      "metadata": {
+        "id": "btI6TD04r3MR"
+      },
+      "execution_count": 34,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "minitrain = MoodBoardDataset(csv_file='minitrain.csv')\n",
+        "\n",
+        "minitests = MoodBoardDataset(csv_file='minitests.csv')"
+      ],
+      "metadata": {
+        "id": "gVnHfbhjsGvH"
+      },
+      "execution_count": 35,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "  class MultiModalEarlyFusionBertModel(BertModel):\n",
+        "    def __init__(self, config, num_classes, cnn_model):\n",
+        "        super().__init__(config)\n",
+        "\n",
+        "        self.cnn_model = torchvision.models.resnet18(pretrained=True)\n",
+        "        # Created f_cnn because we want to run the images through the layers before linear layers\n",
+        "        self.f_cnn = torch.nn.Sequential(*list(resnet_model.children())[:-2])\n",
+        "        self.image_layer= nn.Linear(512, config.hidden_size)\n",
+        "\n",
+        "        # Using a specific pretrained BertModel that was trained for test to emotion purposes with multiclassification\n",
+        "        self.bert_model = AutoModel.from_pretrained(\"ayoubkirouane/BERT-Emotions-Classifier\", output_hidden_states=True)\n",
+        "        self.modality_embedding = nn.Embedding(num_classes, config.hidden_size)\n",
+        "        self.fusion_linear = nn.Linear(config.hidden_size, num_classes)\n",
+        "        self.loss_fct = nn.CrossEntropyLoss()\n",
```

Comment:
It seems like your accuracy results may indicate that your model is overfitting, so it could be a good idea to add a dropout layer or introduce an activation layer for non-linearity for output.